src/geometries/polytopes.jl

```diff
@@ -156,6 +156,16 @@ function angles(c::Chain)
   map(i -> ∠(vs[i - 1], vs[i], vs[i + 1]), i1:i2)
 end
 
+function (c::Chain)(t)
+  if t < 0 || t > 1
+    throw(DomainError(t, "c(t) is not defined for t outside [0, 1]."))
+  end
+  s = collect(segments(c))
```

Comment:
I'm not sure if this is the best way to do it. If there would be an alternative that is not or at least less allocating this would be of course preferable.

Reply:
Would
```julia
  v = vertices(c)
  n = length(v) - !isclosed(c)
  k = max(1, ceil(Int, n * t))
  for (i, seg) in enumerate(segments(c))
    if i == k
      return seg(n * t - k + 1)
    end
  end
```
be better?

Reply:
We can write `s, _ = iterate(segments(c), k)` to retrieve the k-th element of the generator.

Reply:
Ah, that's what I was looking for, but missed. Thanks @juliohm. Implemented in [05a195b](https://github.com/JuliaGeometry/Meshes.jl/pull/1155/commits/05a195be68055a2da3cf8bcd49a970aac5bc1017).